src/msw/webview_edge.cpp

```diff
@@ -0,0 +1,801 @@
+    case wxWEBVIEW_ZOOM_TINY:
+        zoom_factor = 0.5f;
+        break;
+    default:
+        break;
+    }
+    m_impl->m_webView->put_ZoomFactor(zoom_factor);
+}
+
+bool wxWebViewEdge::CanCut() const
+{
+    return false;
+}
+
+bool wxWebViewEdge::CanCopy() const
+{
+    return false;
+}
+
+bool wxWebViewEdge::CanPaste() const
+{
+    return false;
+}
+
+void wxWebViewEdge::Cut()
+{
+
+}
+
+void wxWebViewEdge::Copy()
+{
+
+}
+
+void wxWebViewEdge::Paste()
+{
+
+}
+
+bool wxWebViewEdge::CanUndo() const
+{
+    return false;
+}
+
+bool wxWebViewEdge::CanRedo() const
+{
+    return false;
+}
+
+void wxWebViewEdge::Undo()
+{
+
+}
+
+void wxWebViewEdge::Redo()
+{
+
+}
+
+long wxWebViewEdge::Find(const wxString& WXUNUSED(text), int WXUNUSED(flags))
+{
+    return -1;
+}
+
+//Editing functions
+void wxWebViewEdge::SetEditable(bool WXUNUSED(enable))
+{
+
+}
+
+bool wxWebViewEdge::IsEditable() const
+{
+    return false;
+}
+
+void wxWebViewEdge::SelectAll()
+{
+
+}
+
+bool wxWebViewEdge::HasSelection() const
+{
+    return false;
+}
+
+void wxWebViewEdge::DeleteSelection()
+{
+
+}
+
+wxString wxWebViewEdge::GetSelectedText() const
+{
+    return wxString();
+}
+
+wxString wxWebViewEdge::GetSelectedSource() const
+{
+    return wxString();
+}
+
+void wxWebViewEdge::ClearSelection()
+{
+
+}
+
+void wxWebViewEdge::EnableContextMenu(bool enable)
+{
+    wxCOMPtr<IWebView2Settings2> settings(m_impl->GetSettings());
+    if (settings)
+        settings->put_AreDefaultContextMenusEnabled(enable);
+}
+
+bool wxWebViewEdge::IsContextMenuEnabled() const
+{
+    wxCOMPtr<IWebView2Settings2> settings(m_impl->GetSettings());
+    if (settings)
+    {
+        BOOL menusEnabled = TRUE;
+        settings->get_AreDefaultContextMenusEnabled(&menusEnabled);
+
+        if (!menusEnabled)
+            return false;
+    }
+    return true;
+}
+
+void wxWebViewEdge::EnableDevTools(bool enable)
+{
+    wxCOMPtr<IWebView2Settings2> settings(m_impl->GetSettings());
+    if (settings)
+        settings->put_AreDevToolsEnabled(enable);
+}
+
+bool wxWebViewEdge::IsAccessToDevToolsEnabled() const
+{
+    wxCOMPtr<IWebView2Settings2> settings(m_impl->GetSettings());
+    if (settings)
+    {
+        BOOL devToolsEnabled = TRUE;
+        settings->get_AreDevToolsEnabled(&devToolsEnabled);
+
+        if (!devToolsEnabled)
+            return false;
+    }
+
+    return true;
+}
+
+void* wxWebViewEdge::GetNativeBackend() const
+{
+    return m_impl->m_webView;
+}
+
+bool wxWebViewEdge::RunScriptSync(const wxString& javascript, wxString* output)
+{
+    bool scriptExecuted = false;
+
+    HRESULT hr = m_impl->m_webView->ExecuteScript(javascript.wc_str(), Callback<IWebView2ExecuteScriptCompletedHandler>(
+        [&scriptExecuted, output](HRESULT error, PCWSTR result) -> HRESULT
+    {
+        if (error == S_OK)
+        {
+            if (output)
+                output->assign(result);
+        }
+        else
+            wxLogError(_("RunScript failed: %.8x"), error);
```

Comment:
AFAICS this method is supposed to fill `output` with the error when it fails, and it's logged by the code in the caller. So this should be something like

```suggestion
            output->Printf("%s (0x%08lx)", wxSysErrorMsgStr(error), error);
            return false;
```